accepted/00000-remote-command-auditing.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: Remote Command Auditing
+- Start Date: 2019-06-21
+- RFC PR: TBD
+
+# Summary
+[summary]: #summary
+
+Implement a role-based access model to access `Remote Command` for managed systems and log every command executed.
+
+# Motivation
+[motivation]: #motivation
+
+At the time of the writing, there are three features to issue a remote command on a managed system or a group of systems registered to the product:
+- A `Remote Command` tab is present in the details section of a traditional or minion system (or a `System Set Manager` group)
+- The API exposes the `system.scheduleScriptRun` to issue a command
+- (Only for minions) a `Salt > Remote Commands` page to launch a command on minions by leveraging [Salt targeting](https://docs.saltstack.com/en/latest/topics/targeting/index.html) to identify the target(s)
+
+Each user can only target the systems within the same organization that the systems are registered to, provided that the user has a particular role:
+- Every user that has `SUSE Manager Administrator` or `Organization Administrator` role can access the tab and run a remote command on the selected system or a group of systems
+- The API is always exposed but checks for the role of the user supplying the command after the API is called
+- For every user registered to SUSE Manager, independent of the role, the menu item `Salt > Remote Commands` is accessible. Only users with the `SUSE Manager Administrator` or `Organization Administrator` role, though, can impact systems using this feature, whereas all other users cannot target any system (`"no systems found"`).
+
+Every system outside the organization of the user issuing the command is not impacted nor is it visible by the user.
+
+The motivations behind the changes introduced in this RFC are _security_ and _auditing_. Any command issues via the `Remote Command` will be run with `root` privileges on the target system(s). Additionally, the user that issued the command is not logged along with the command issues and its output.
+Many customers claim to require this feature before migrating from traditional clients to Salt minions.
+
+Reference: https://fate.suse.com/325624 https://fate.suse.com/327158 
+
+In order to overcome the above limitations, in this RFC we are going to introduce:
+
+- Role-based restriction for `Remote Command`: a new role will be introduced. Every user associated with this role (in addition to one of the `Administrative Role`s) can access the `Remote Command` feature and issue commands. Every other user will not see the `Remote Command` tab nor the `Salt > Remote Commands` menu item.
```

Comment:
`(in addition to one of the 'Administrative Roles')`

Does it mean beeing an Org Admin would not be enough to execute remote commands? 

This is not typicall. An Org Admin (or SUSE Manager Administrator) are like "root". They should be allowed to do everything. Indirectly all roles are assigned to them.

As I know people requested this extra role I wonder if they do not want to achieve something different.
Maybe they want to allow user who are not Org Admin to also access Remote Command page for a limited number of systems.
I think we should ask @mbrookhuis and Don again to explain what customers really want.

Reply:
> `(in addition to one of the 'Administrative Roles')`
> 
> Does it mean beeing an Org Admin would not be enough to execute remote commands?

It is in the current situation (as-is). But in the future, for new users, you will have to manually set the new role `Remote Command Administrator` to that user. Existing users will behave with the old behavior, as requested by champions.
I pushed a clarification on this regard.

> As I know people requested this extra role I wonder if they do not want to achieve something different.
> Maybe they want to allow user who are not Org Admin to also access Remote Command page for a limited number of systems.

In this case the `Remote Command Administrator` role should be then an `Administrative Role`, equal to the  `SUSE Manager Administrator` or `Organization Administrator roles` in order to target every system within the same organization as the user issuing the command.

Reply:
Added a clarifying section: `Administrative Roles` will not be impacted. They will remain `root`.